### PR TITLE
Jon/android walletconnect modal fix

### PR DESCRIPTION
**Pull request by jonandgon**
## Description
This PR fixes the walletconnect modal on android devices. 

it also removes the need for https://walletconnect.puzzle.online

## Ticket
n/a

## Steps to test
- run `pnpm i` then `pnpm build` on the sdk
- point devtools to this PR
- run `pnpm i` then `pnpm --filter devtools` on devtools
  - test devtools on the android emulator
  - test devtools on the ios emulator
  - test devtools on desktop

## Technicals
we're patching `@walletconnect/modal-core` and `@walletconnect/modal-ui` to achieve this. there's some small walletconnect bs that differentiates android and mobile for some reason. We also patch it to directly send the wc params to the extension

## Associated PRs
review + merge #29 first pls :)

### CODE REVIEW

**Review by lukenewman — APPROVED** [review]
looks great -- able to connect android wallet to local (desktop) devtools -- web3modal still didn't show puzzle wallet as option when pointed to https://puzzle-sdk-core-example.vercel.app/ on the device, but tapping "Select Wallet" opened the app and prompted connection -- now going back to the core example site and tapping connect is auto-connecting without showing modal or prompt, even when opening new tabs to that example site

File: packages/core/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@puzzlehq/sdk-core",
   "displayName": "Puzzle SDK",
-  "version": "0.3.0",
+  "version": "0.2.31-beta.2",
```
[the remark this answers is not in the document]
Comment (jonandgon) [reply packages/core/package.json:4]: yeah will make it 0.3.0-alpha.1 before merging

[the remark this answers is not in the document]
Comment (lukenewman) [reply packages/core/package.json]: @jonandgon bad choice of words on my part -- i meant something like `0.3.0` or any other non-suffixed version

[the remark this answers is not in the document]
Comment (jonandgon) [reply packages/core/package.json]: oops yes haha i can do that rn

**Pull request merged**